awx/ui_next/src/components/Lookup/Lookup.test.jsx

```diff
@@ -44,348 +42,118 @@ async function checkInputTagValues(wrapper, expected) {
-  );
-  expect(chips).toHaveLength(expected.length);
-  chips.forEach((el, index) => {
-    expect(el.text()).toEqual(expected[index]);
-  });
-}
+const QS_CONFIG = getQSConfig('test', {});
+const TestList = () => <div />;
 
-describe('<Lookup multiple/>', () => {
+describe('<Lookup />', () => {
   let wrapper;
   let onChange;
 
-  beforeEach(() => {
+  async function mountWrapper() {
     const mockSelected = [{ name: 'foo', id: 1, url: '/api/v2/item/1' }];
+    await act(async () => {
+      wrapper = mountWithContexts(
+        <Lookup
+          id="test"
+          multiple
+          header="Foo Bar"
+          value={mockSelected}
+          onChange={onChange}
+          qsConfig={QS_CONFIG}
+          renderOptionsList={({ state, dispatch, canDelete }) => (
+            <TestList
+              id="options-list"
+              state={state}
+              dispatch={dispatch}
+              canDelete={canDelete}
+            />
+          )}
+        />
+      );
+    });
+    return wrapper;
+  }
+
+  beforeEach(() => {
     onChange = jest.fn();
     document.body.innerHTML = '';
-    wrapper = mountWithContexts(
-      <Lookup
-        multiple
-        lookupHeader="Foo Bar"
-        name="foobar"
-        value={mockSelected}
-        onLookupSave={onChange}
-        getItems={() => ({
-          data: {
-            count: 2,
-            results: [
-              ...mockSelected,
-              { name: 'bar', id: 2, url: '/api/v2/item/2' },
-            ],
-          },
-        })}
-        columns={mockColumns}
-        sortedColumnKey="name"
-      />
-    );
   });
 
-  test('Initially renders succesfully', () => {
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  test('should render succesfully', async () => {
+    wrapper = await mountWrapper();
     expect(wrapper.find('Lookup')).toHaveLength(1);
   });
 
-  test('Expected items are shown', async done => {
+  test('should show selected items', async () => {
+    wrapper = await mountWrapper();
     expect(wrapper.find('Lookup')).toHaveLength(1);
     await checkInputTagValues(wrapper, ['foo']);
-    done();
   });
 
-  test('Open and close modal', async done => {
-    checkRootElementNotPresent('body div[role="dialog"]');
-    wrapper.find('button[aria-label="Search"]').simulate('click');
-    checkRootElementPresent('body div[role="dialog"]');
-    // This check couldn't pass unless api response was formatted properly
-    await checkModalListValues(wrapper, [[true, 'foo'], [false, 'bar']]);
-    wrapper.find('Modal button[aria-label="Close"]').simulate('click');
+  test('should open and close modal', async () => {
+    wrapper = await mountWrapper();
     checkRootElementNotPresent('body div[role="dialog"]');
     wrapper.find('button[aria-label="Search"]').simulate('click');
     checkRootElementPresent('body div[role="dialog"]');
+    const list = wrapper.find('TestList');
+    expect(list).toHaveLength(1);
+    expect(list.prop('state')).toEqual({
```

Comment:
@keithjgrant @mabashian 
I realize I'm nitpicking this PR at the last second, but can we test the external behavior of the component here and not it's internal state? I think that would make the tests more robust.

Reply:
oh, I just saw this after sticking `mergeit` on there

Yeah, I agree in general... this was something that was there before my changes & I was eager to get the merge in so I left it. But it's worth flagging for later

Reply:
hmm, I'm referring to the `list.prop('state')` check, which I don't currently see on devel :confused: